eth/blockwatch/block_watcher.go

```diff
@@ -325,7 +333,9 @@ func (w *Watcher) getMissedEventsToBackfill(ctx context.Context) ([]*Event, erro
 	if err != nil {
 		return events, err
 	}
-	latestBlockNum := int(latestBlock.Number.Int64())
+
+	// Latest block will be polled separately in syncToLatestBlock(), so it's not backfilled.
+	preLatestBlockNum := int(latestBlock.Number.Int64()) - 1
```

Comment:
Is there any reason behind omitting the latest block? Is it to always treat the latest block as a new one?

Reply:
It's tricky, I tried to explain it here: https://github.com/livepeer/go-livepeer/pull/2489/files#diff-2f2752a740f61bfa5688f822d8850e6d751507520b6c28198afc60f35b8f90b3R89.

In short, we watch blocks for 2 reasons:
- Updating local cache => backfilling in batches is good enough
- Timing => Passing the block to the TimeWatcher, then we need to poll the block separately, because even if the batch does not contain the interesting event, we need to pass it to TimeWatcher